client/features/authSlice.tsx

```diff
@@ -0,0 +1,30 @@
+import { createSlice } from '@reduxjs/toolkit'
+import type { PayloadAction } from '@reduxjs/toolkit'
+import type { User } from '../../src/types/custom'
+import type { RootState } from '../app/store'
+
+type AuthState = {
+  user: User | null
+  token: string | null
+}
+
+const authSlice = createSlice({
+  name: 'auth',
+  initialState: { user: null, token: null } as AuthState,
+  reducers: {
+    setCredentials: (
+      state,
+      { payload: { user, token } }: PayloadAction<{ user: User; token: string }>
+    ) => {
+      state.user = user
+      state.token = token
+    },
+  },
+})
+
+export const { setCredentials } = authSlice.actions
+
+export default authSlice.reducer
+
+export const selectCurrentUser = (state: RootState) => state.auth.user
```

Comment:
nit: Personally, I don't really prefer these kinds of helpers, I prefer to manually write out the dependencies directly in the component when calling useSelector because I think it makes the code more explicit, but this is a matter of style preference, so it's up to you